pkg/templates/loader/terraform_loader_test.go

```diff
@@ -18,13 +20,187 @@ var mockInfo = &openapi3.Info{
+													Default:  []any{p1},
+													Items: &openapi3.SchemaRef{
+														Value: &openapi3.Schema{
+															Type: openapi3.TypeObject,
+															Required: []string{
+																"email",
+																"name",
+															},
+															Properties: map[string]*openapi3.SchemaRef{
+																"name": {
+																	Value: &openapi3.Schema{
+																		Title: "Name",
+																		Type:  openapi3.TypeString,
+																		Extensions: openapi.NewExt().
+																			WithOriginalType(cty.String).
+																			Export(),
+																	},
+																},
+																"email": {
+																	Value: &openapi3.Schema{
+																		Title:    "Email",
+																		Type:     openapi3.TypeObject,
+																		Required: []string{"domain"},
+																		Properties: map[string]*openapi3.SchemaRef{
+																			"domain": {
+																				Value: &openapi3.Schema{
+																					Title: "Domain",
+																					Type:  openapi3.TypeString,
+																					Extensions: openapi.NewExt().
+																						WithOriginalType(cty.String).
+																						Export(),
+																				},
+																			},
+																		},
+																		Extensions: openapi.NewExt().
+																			WithOriginalType(cty.Object(
+																				map[string]cty.Type{
+																					"domain": cty.String,
+																				})).
+																			WithUIColSpan(12).
+																			Export(),
+																	},
+																},
+															},
+															Extensions: openapi.NewExt().WithOriginalType(
+																cty.Object(
+																	map[string]cty.Type{
+																		"name": cty.String,
+																		"email": cty.Object(map[string]cty.Type{
+																			"domain": cty.String,
+																		}),
+																	})).
+																WithUIColSpan(12).
+																Export(),
+														},
+													},
+													Extensions: openapi.NewExt().
+														WithOriginalType(
+															cty.Tuple(
+																[]cty.Type{
+																	cty.Object(
+																		map[string]cty.Type{
+																			"name": cty.String,
+																			"email": cty.Object(map[string]cty.Type{
+																				"domain": cty.String,
+																			}),
+																		}),
+																})).
+														WithUIGroup("Basic").
+														WithUIColSpan(12).
+														WithUIOrder(3).
+														Export(),
+												},
+											},
+										},
+										Extensions: openapi.NewExt().
+											WithOriginalVariablesSequence([]string{
+												"def_list_string",
+												"def_list_tuple",
+												"def_list_object",
+											}).
+											WithUIGroupOrder("Basic").
+											Export(),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectedError: false,
```

Comment:
can we describe the data in YAML(to ensure the structure), and then marshal it as openapi schema before each test run?